### PR TITLE
[RHCLOUD-34615] Cronjob to automatically generate clients and open a PR when API change is available

### PR DESCRIPTION
**Pull request by tahmidefaz**
`alpha` tagging with `jscutlery/semver` nx plugin: [Here](https://github.com/jscutlery/semver?tab=readme-ov-file#specify-the-level-of-change)

Here is how the workflow will look like:
1. The new github action will run once a day.
2. If new changes to API specs are detected, it will generate clients and open a PR with it.
3. Engineers will quickly review the changes and merge the PR if everything looks good.
4. Once merged, the `release-alpha` action will run and the packages will be released with a `prerelease` `alpha` tag.

### CODE REVIEW

**Review by florkbr — COMMENTED** [review]
@tahmidefaz I think we have 2 paths to take this:
1. If we keep this approach - I think we will need to create a separate branch (such as `nightly`) - otherwise the pull request the bot opens will target main and in that pull request we'd have each client as an `alpha` version in the package.json. If we create a new branch, we can just commit directly to that branch in lieu of a pull request (though we'd need to worry about keeping it up to date with main)
2. Lower hanging fruit would be to have a cron job that runs nightly and opens a PR against master (if packages changed) with no version bump. And then we can merge and rely on our existing automation to publish a patch version

File: .github/workflows/sync-apis.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      run: npm run build:no-cache
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v4
+      with:
+        token: ${{ secrets.BOT_GH_TOKEN }}
+        title: Syncing APIs and Generating Clients
+        body: |
+          There were some changes in the contents of the APIs.
+
+          Automated changes by [create-pull-request](https://github.com/peter-evans/create-pull-request) GitHub action
```
Comment (florkbr) [inline .github/workflows/sync-apis.yml:48]: I don't think we need this comment; maybe just add that this was generated by a cron job?

File: .github/workflows/sync-apis.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        commit-message: Syncing API
```
Comment (florkbr) [inline .github/workflows/sync-apis.yml:49]: @tahmidefaz FYI this commit message differs from what you expect above in the workflow

See https://github.com/RedHatInsights/javascript-clients/pull/300/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR72

We may also want to consider using convention commit syntax for this so we avoid someone making the same commit message on accident (something like chore: or release: ). See https://www.conventionalcommits.org/en/v1.0.0/#summary

Comment (tahmidefaz) [reply .github/workflows/sync-apis.yml]: Done, replaced it with a `chore` prefix.

**Comment by tahmidefaz** [comment]
@florkbr we choose to go with option 2. It also allowed me to remove the `release-alpha` workflows. Now, it's just a simple cron job that will run nightly, and check for updates to APIs, generate clients accordingly, and open a PR afterwards with those changes.

**Review by florkbr — APPROVED** [review]

**Comment by florkbr** [comment]
Looks good to me. I still think we should follow up with a script to publish nightly to npm. I'll whip that up as a follow up (we still want this automated PR regardless).

**Pull request merged**